relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,105 @@
+use std::borrow::Cow;
+use std::sync::Arc;
+use std::time::Duration;
+
+use relay_dynamic_config::GlobalConfig;
+use relay_system::{Addr, Interface, Service};
+use reqwest::Method;
+use serde::{Deserialize, Serialize};
+
+use crate::actors::project_cache::ProjectCache;
+use crate::actors::upstream::{RequestPriority, SendQuery, UpstreamQuery, UpstreamRelay};
+
+// No messages are accepted for now.
+#[derive(Debug)]
+pub enum GlobalConfigMessage {}
+
+impl Interface for GlobalConfigMessage {}
+
+/// Service implementing the [`GlobalConfig`] interface.
+///
+/// The service is responsible to fetch the global config appropriately and
+/// forward it to the services that require it.
+#[derive(Debug)]
+pub struct GlobalConfigService {
+    project_cache: Addr<ProjectCache>,
+    upstream: Addr<UpstreamRelay>,
+}
+
+impl Interface for GetGlobalConfig {}
+
+impl GlobalConfigService {
+    pub fn new(project_cache: Addr<ProjectCache>, upstream: Addr<UpstreamRelay>) -> Self {
+        Self {
+            project_cache,
+            upstream,
+        }
+    }
+
+    /// Forwards the given global config to the services that require it.
+    fn update_global_config(&mut self) {
+        let upstream_relay: Addr<UpstreamRelay> = self.upstream.clone();
+        let project_cache = self.project_cache.clone();
+        tokio::spawn(async move {
+            let query = GetGlobalConfig;
+
+            if let Ok(Ok(response)) = upstream_relay.send(SendQuery(query)).await {
+                let global_config = Arc::new(response.global);
+                project_cache.send::<Arc<GlobalConfig>>(global_config);
+            };
+        });
+    }
+}
+
+impl Service for GlobalConfigService {
+    type Interface = GlobalConfigMessage;
+
+    fn spawn_handler(mut self, _rx: relay_system::Receiver<Self::Interface>) {
+        tokio::spawn(async move {
+            let mut ticker = tokio::time::interval(Duration::from_secs(10));
+            relay_log::info!("global config started");
+
+            loop {
+                tokio::select! {
+                    biased;
+                    _ = ticker.tick() => self.update_global_config(),
+                    else => break,
+                }
+            }
+            relay_log::info!("global config stopped");
+        });
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfig;
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfigResponse {
+    global: GlobalConfig,
+}
+
+impl UpstreamQuery for GetGlobalConfig {
+    type Response = GetGlobalConfigResponse;
+
+    fn method(&self) -> Method {
+        Method::POST
+    }
+
+    fn path(&self) -> Cow<'static, str> {
+        Cow::Borrowed("/api/0/relays/projectconfigs/?version=4&global_config=true")
```

Comment:
should match my sentry side change: https://github.com/getsentry/sentry/pull/52421/files#diff-1f1d2ba0fbdd84250e074bd0ec21f8269957b5128716d2e6c3774d705f73d98dR54

Reply:
Not sure what the current state of the proposal is, but in the last discussion with @olksdr and @iker-barriocanal we decided to put the request for `"global": true` in the request body (not as a query parameter).